label_studio_ml/examples/mmdetection-3/mmdetection.py

```diff
@@ -0,0 +1,152 @@
+        config_file = config_file or os.environ['config_file']
+        checkpoint_file = checkpoint_file or os.environ['checkpoint_file']
+        self.config_file = config_file
+        self.checkpoint_file = checkpoint_file
+        self.labels_file = labels_file
+        # default Label Studio image upload folder
+        upload_dir = os.path.join(get_data_dir(), 'media', 'upload')
+        self.image_dir = image_dir or upload_dir
+        logger.debug(f'{self.__class__.__name__} reads images from {self.image_dir}')
+        if self.labels_file and os.path.exists(self.labels_file):
+            self.label_map = json_load(self.labels_file)
+        else:
+            self.label_map = {}
+
+        self.from_name, self.to_name, self.value, self.labels_in_config = get_single_tag_keys(
+            self.parsed_label_config, 'RectangleLabels', 'Image')
+        schema = list(self.parsed_label_config.values())[0]
+        self.labels_in_config = set(self.labels_in_config)
+
+        # Collect label maps from `predicted_values="airplane,car"` attribute in <Label> tag
+        self.labels_attrs = schema.get('labels_attrs')
+        if self.labels_attrs:
+            for label_name, label_attrs in self.labels_attrs.items():
+                for predicted_value in label_attrs.get('predicted_values', '').split(','):
+                    self.label_map[predicted_value] = label_name
+
+        print('Load new model from: ', config_file, checkpoint_file)
+        self.model = init_detector(config_file, checkpoint_file, device=device)
+        self.score_thresh = score_threshold
+
+
+    def _get_image_url(self, task):
+        image_url = task['data'].get(self.value) or task['data'].get(DATA_UNDEFINED_NAME)
+        if image_url.startswith('s3://'):
+            # presign s3 url
+            r = urlparse(image_url, allow_fragments=False)
+            bucket_name = r.netloc
+            key = r.path.lstrip('/')
+            client = boto3.client('s3')
+            try:
+                image_url = client.generate_presigned_url(
+                    ClientMethod='get_object',
+                    Params={'Bucket': bucket_name, 'Key': key}
+                )
+            except ClientError as exc:
+                logger.warning(f'Can\'t generate presigned URL for {image_url}. Reason: {exc}')
+        return image_url
+
+    def predict(self, tasks, **kwargs):
+        assert len(tasks) == 1
+        task = tasks[0]
+        image_url = self._get_image_url(task)
+        image_path = self.get_local_path(image_url)
+        model_results = inference_detector(self.model, image_path).pred_instances
+        results = []
+        all_scores = []
+        img_width, img_height = get_image_size(image_path)
+        print(f">>> model_results: {model_results}")
+        print(f">>> label_map {self.label_map}")
+        print(f">>> self.model.dataset_meta: {self.model.dataset_meta}")
+        classes = self.model.dataset_meta.get('CLASSES')
```

Comment:
The keys in the dataset meta have been refactored to lowercase after https://github.com/open-mmlab/mmdetection/pull/9469. The 3.0.0rc5 will be released today and this change will take effect.

Reply:
Fixed